proposed/0006-Launcher.md

```diff
@@ -0,0 +1,204 @@
+
+**step 1:** All `GET/PUT/POST/DELETE`s will go via the Launcher. The App will essentially encrypt using `<App-Specific-Symm-Key>` and pass it to Launcher Background Process which will decrypt and Re-encrypt and sign it using normal process (currently MAID-keys). For `GET`s it will be the reverse - Launcher will eventually encrypt using `<App-Specific-Symm-Key>` before handing the data over to the App.
+
+**step 2:** Since `<App-Specific-Symm-Key>` is recognised by Launcher only for current session, there is no security risk and the App will not be able to trick Launcher the next time it starts to use the previous keys to mutate network or read data on its behalf.
+
+## Share Directory App Flow
+
+Every time the App tries to access `SAFEDrive` Launcher will check the permission in `<MAIDSAFE-SPECIFIC-CONFIG-ROOT>/LauncherReservedDirectory/LauncherConfigurationFile`.
+### Grant Access
+- User can grant an App's access of `SAFEDrive` for shared access. Launcher will simply add this information to `<MAIDSAFE-SPECIFIC-CONFIG-ROOT>/LauncherReservedDirectory/LauncherConfigurationFile`.
+
+### Revoke Access
+- User can revoke an App's access of `SAFEDrive` for shared access. Launcher will simply remove this information from `<MAIDSAFE-SPECIFIC-CONFIG-ROOT>/LauncherReservedDirectory/LauncherConfigurationFile`.
+
+## Remove App Flow
+
+**procedure 0:** Launcher removes the App as follows:
+- Delete from `<LOCAL-CONFIG-FILE>` (on the user's machine) the following:
+```
+[
+    { App-ID, “/path/to/XYZ” }, // Remove this, other entries are untouched.
+    ...
+]
+```
+- Remove the SHA512(App-Binary) from the vector in `<LAUNCHER-CONFIG-FILE>`.
+- Decrement `Reference Count` from `<LAUNCHER-CONFIG-FILE>`.
+- If the `Reference Count` is **0** it means that this is the last machine where the App was present. The Launcher shall not delete `<APP-ROOT-DIR>` from within `SAFEDrive` folder. It is user's responsibility to do that as it might contain information (like pictures etc) which the user may not want to lose. Instead the Launcher will ask if the user wants to do that and act accordingly. Similary only after user confirmation will Launcher remove the App entry from the `<LAUNCHER-CONFIG-FILE>`, as it contains necessary metadata to decrypt the App specific directories.
+
+**procedure 1:** While the other procedure would work, but there might be occassions when the user wants to immediately remove the app completely (which also translates as revoke App's permission to mutate network on user's behalf). He may not have access to other machines where the App was installed and maybe currently running and the previous procedure requires him to remove it from all machines. Thus there shall be an option in Launcher to remove App completely irrespective of if it is installed and/or running in other machines. In such cases Launcher will reduce `Reference Count` to 0 and proceed as above for the detection of zero reference count.
```

Comment:
"procedure would work, but there" -> "procedure would work, there"
Replace  "he", "him", etc. with "user"
"maybe" -> "may be"